### PR TITLE
Updating the binary search example

### PR DESCRIPTION
**Pull request by konnov**
So far, we have not been testing the binary search example. This PR fixed that.

- [x] Tests added for any new code
- [ ] Entries added to the respective `CHANGELOG.md` for any new functionality

### CODE REVIEW

File: examples/classic/sequential/BinSearch/BinSearch.qnt [hunk 1]
```diff
@@ -2,7 +2,7 @@
 /**
  * A tutorial version of binary search. The original TLA+ specification:
  *
- * https://github.com/informalsystems/apalache/blob/unstable/test/tla/bin-search/BinSearch10.tla
+ * https://github.com/informalsystems/apalache/blob/unstable/test/tla/bin-search/BinSearch.tla
```
Comment (thpani) [inline examples/classic/sequential/BinSearch/BinSearch.qnt:5]: The link is broken,, probably also in other files?

**Review by thpani — APPROVED** [review]
Looks good, but the link you replaced is still broken?

File: examples/classic/sequential/BinSearch/BinSearch.qnt (hunk 1 above)
Comment (thpani) [reply examples/classic/sequential/BinSearch/BinSearch.qnt:5]: Also note that `unstable` should probably be `main`?

Comment (konnov) [reply examples/classic/sequential/BinSearch/BinSearch.qnt:5]: ah, all true :)

Comment (konnov) [inline examples/classic/sequential/BinSearch/BinSearch.qnt:5]:
```suggestion
 * https://github.com/informalsystems/apalache/blob/main/test/tla/bin-search/BinSearch10.tla
```

**Pull request merged**